tests/integration/extenders/NotificationChannelTest.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Tests\integration\extenders;
+
+use Flarum\Extend\NotificationChannel;
+use Flarum\Tests\integration\RetrievesAuthorizedUsers;
+use Flarum\Tests\integration\TestCase;
+use Flarum\User\User;
+
+class NotificationChannelTest extends TestCase
```

Comment:
Following in the convention for extender tests so far, I'd like to see at least tests for:

User::getPreferences does not contain a channel if it has not been added via an extender
A given type is disabled/null (what should it be by the way if it hasn't been registered). Also should there be a way to register types? Anyway, a given type is disabled/null if it hasn't been enabled/disabled.
User::getPreferences DOES contain a channel if it has been added via an extender
A given type is enabled for a user that hasn't overriden that if it has been enabled by default via an extender
A user can override a default type preference by setting their own preference.